frontend/src/metabase/actions/containers/ActionCreator/ActionCreator.unit.spec.tsx

```diff
@@ -120,5 +163,104 @@ describe("ActionCreator", () => {
 
       expect(screen.getByText("FooBar")).toBeInTheDocument();
     });
+
+    describe("admin users and has public sharing enabled", () => {
+      const mockUuid = "mock-uuid";
+
+      it("should show action settings button", async () => {
+        await setupEditing({
+          isAdmin: true,
+          isPublicSharingEnabled: true,
+        });
+
+        expect(getIcon("gear", "button")).toBeInTheDocument();
+      });
+
+      it("should be able to enabled action public sharing", async () => {
+        await setupEditing({
+          isAdmin: true,
+          isPublicSharingEnabled: true,
+        });
+
+        getIcon("gear", "button").click();
+
+        expect(screen.getByText("Action settings")).toBeInTheDocument();
+        expect(
+          screen.getByRole("switch", { name: "Make public" }),
+        ).not.toBeChecked();
+        expect(
+          screen.queryByRole("textbox", { name: "Public action link URL" }),
+        ).not.toBeInTheDocument();
+
+        // Make the action public
+        screen.getByRole("switch", { name: "Make public" }).click();
+
+        await waitFor(() => {
+          expect(
+            screen.getByRole("switch", { name: "Make public" }),
+          ).toBeChecked();
+        });
```

Comment:
Let's use `await screen.findByRole` instead of `waitFor` here

Reply:
afair it doesn't work but I'll give it another try. `await screen.findByRole` will get you an element. And since the elements is already on screen the test will fail since we haven't waited for it to turn unchecked.

From my experience, uses `await screen.findBy` to get element that isn't on screen yet. And use `waitFor` if you expects existing elements to change in another way.

Reply:
This confirms it doesn't work.
<img width="1361" alt="image" src="https://user-images.githubusercontent.com/1937582/215116098-545449f2-5390-423c-b383-6895c2bc88e7.png">